### DISCUSSION TITLE
How to push two routes to the stack without flickering transition

### DISCUSSION THREAD

**Discussion by ivan-unfolds** [react-navigation/react-navigation#10171] [Q&A]
Hi I'm trying to use React Navigation to navigate directly from screen A to screen C whilst at the same time programmatically inserting screen B in between them so that when I use the `Back`  button on screen C I end up in screen B rather than screen A. I've got it working using the following code 

```jsx
navigation.push('B')
navigation.push('C')
```

The only problem is that during the transition from A to C, screen B briefly flickers into view. I'd like to get rid of this flickering transition so that the it looks like a direct transition from A to C, whilst imperceptibly adding screen B in between them. Is there any way to do this?

 I also tried using `navigation.reset` but had the same transition issue as when using `navigation.push`. 

```jsx
navigation.reset({
    index: 2,
    routes: [
      {name: 'A'},
      {name: 'B'},
      {name: 'C'},
    ],
  })
```

**Comment by satya164** [comment]
> I also tried using navigation.reset

That's destroying the whole state and setting it to a new one. To add a new route to existing state, you need to pass a function to `dispatch` and update the `routes` array: https://reactnavigation.org/docs/navigation-actions#reset
